connectors/api/src/main/java/com/datastax/oss/dsbulk/connectors/api/CommonConnectorFeature.java

```diff
@@ -28,4 +28,10 @@
    * strings).
    */
   MAPPED_RECORDS,
+
+  /**
+   * Indicates that the connector supports data sampling for the performance optimization purpose.
+   * It should return not supported if the data source cannot be read more than once.
```

Comment:
```suggestion
   * Data size sampling is typically done when using the connector for reading, in which case a sample of the data  is read before the actual read operation begins. If the data source cannot be read more than once, then data size sampling should be disallowed. This is notably the case when reading live data streams such as {@linkplain System#in standard input}.
```

(you will have to reformat the javadocs)